apps/docs/content/guides/local-development/testing/pgtap-extended.mdx

```diff
@@ -0,0 +1,653 @@
+  name text not null,
+  slug text unique not null,
+  plan_type text not null check (plan_type in ('free', 'pro', 'enterprise')),
+  max_posts int not null default 5,
+  created_at timestamptz default now()
+);
+
+create table public.org_members (
+  org_id bigint references public.organizations(id) on delete cascade,
+  user_id uuid references auth.users(id) on delete cascade,
+  role text not null check (role in ('owner', 'admin', 'editor', 'viewer')),
+  created_at timestamptz default now(),
+  primary key (org_id, user_id)
+);
+
+create table public.posts (
+  id bigint primary key generated always as identity,
+  title text not null,
+  content text not null,
+  author_id uuid references public.profiles(id) not null,
+  org_id bigint references public.organizations(id),
+  status text not null check (status in ('draft', 'published', 'archived')),
+  is_premium boolean default false,
+  scheduled_for timestamptz,
+  category text,
+  view_count int default 0,
+  published_at timestamptz,
+  created_at timestamptz default now(),
+  updated_at timestamptz default now()
+);
+
+create table public.comments (
+  id bigint primary key generated always as identity,
+  post_id bigint references public.posts(id) on delete cascade,
+  author_id uuid references public.profiles(id),
+  content text not null,
+  is_deleted boolean default false,
+  created_at timestamptz default now(),
+  updated_at timestamptz default now()
+);
+```
+
+#### 2. RLS policies declaration
+
+Now we want to setup the RLS policies for each of our tables:
+
+```sql
+-- Helper function for role checks
+create or replace function public.get_user_org_role(org_id bigint, user_id uuid)
+returns text as $$
+  select role from org_members
+  where org_id = $1 and user_id = $2;
+-- Note the use of security definer to avoid RLS checking recursion issue
+-- see: https://supabase.com/docs/guides/database/postgres/row-level-security#use-security-definer-functions
+$$ language sql security definer;
+
+-- Enable RLS for all tables
+alter table public.profiles enable row level security;
+alter table public.organizations enable row level security;
+alter table public.org_members enable row level security;
+alter table public.posts enable row level security;
+alter table public.comments enable row level security;
+
+-- Profiles policies
+create policy "Public profiles are viewable by everyone"
+  on public.profiles for select using (true);
+
+create policy "Users can insert their own profile"
+  on public.profiles for insert with check (auth.uid() = id);
+
+create policy "Users can update their own profile"
+  on public.profiles for update using (auth.uid() = id)
+  with check (auth.uid() = id);
+
+-- Organizations policies
+create policy "Public org info visible to all"
+  on public.organizations for select using (true);
+
+create policy "Org management restricted to owners"
+  on public.organizations for all using (
+    public.get_user_org_role(id, auth.uid()) = 'owner'
```

Comment:
```suggestion
    public.get_user_org_role(id, (select auth.uid())) = 'owner'
```

not 100% sure of the syntax for this one, you might wanna check, but I _think_ this should trigger the initPlan optimization